### PR TITLE
Improve performance

### PR DESCRIPTION
**Pull request by jafyvilla**
Modify memory layout in EventSource so computations can be
vectorized for performance.
Store all the events in a single array, memory aligned.
Invert the radius of the events at loading time, to speedup the
computation.

### CODE REVIEW

File: fivox/eventSource.h [hunk 1]
```diff
@@ -70,14 +98,21 @@ class EventSource
...
-
-    /** Add a new event and update the bounding box. Not thread safe. */
-    void add( const Event& event );
+    /**
+     * Update attributes of the event specified by the index. Update also the
+     * bounding box to include the new position. The specified index should
+     * be smaller than the size used in resize()
+     * Not thread safe.
+     */
+    void update( const size_t i, const Vector3f pos,
```
Comment (tribal-tec) [inline fivox/eventSource.h]: const& and rm const for copy-by-values

Comment (tribal-tec) [inline fivox/eventSource.h]: doxygen for params?

File: fivox/fieldFunctor.h [hunk 2]
```diff
@@ -50,40 +50,44 @@ FieldFunctor< TImage >::operator()( const TPoint& point, const TSpacing& ) const
...
-
     const float cutOffDistance = Super::_source->getCutOffDistance();
-    const AABBf region( base - Vector3f( cutOffDistance ),
-                        base + Vector3f( cutOffDistance ));
-    const Events& events = Super::_source->findEvents( region );
 
-    const float squaredCutoff = cutOffDistance * cutOffDistance;
-    float sum = 0;
-    for( const Event& event : events )
+    const size_t size = Super::_source->getNumEvents();
```
Comment (tribal-tec) [inline fivox/fieldFunctor.h]: move parts of the comment from lfpFunctor to here to explain this __restrict__, vector aligned magic here?

File: fivox/synapseLoader.cpp [hunk 3]
```diff
@@ -45,17 +45,22 @@ class SynapseLoader::Impl
...
         for( const uint32_t gid : gids )
         {
             const brion::SynapseMatrix& data =
                 synapses.read( gid, brion::SYNAPSE_PRESYNAPTIC_SURFACE_X |
                                     brion::SYNAPSE_PRESYNAPTIC_SURFACE_Y |
                                     brion::SYNAPSE_PRESYNAPTIC_SURFACE_Z );
             for( size_t i = 0; i < data.shape()[0]; ++i )
-                _output.add( Event( Vector3f( data[i][0], data[i][1],
-                                              data[i][2] ), 1.f ));
+                positions.push_back(
```
Comment (tribal-tec) [inline fivox/synapseLoader.cpp]: emplace_back

File: fivox/types.h [hunk 4]
```diff
@@ -57,7 +57,8 @@ typedef itk::Image< float, 3 > FloatVolume;
 typedef std::shared_ptr< EventFunctor< ByteVolume >> ByteFunctorPtr;
 typedef std::shared_ptr< EventFunctor< FloatVolume >> FloatFunctorPtr;
 
-typedef std::vector< Event > Events;
+typedef std::unique_ptr< float > Events;
+typedef std::vector< float > EventValues;
```
Comment (tribal-tec) [inline fivox/types.h]: use the floats typedef from Brion?

**Comment by timocafe** [comment]
miaou !

**Comment by jafyvilla** [comment]
Comments addressed, including the miaou

**Comment by tribal-tec** [comment]
+1 for me

**Pull request merged**